app/src/main/java/org/mozilla/fenix/components/TabCollectionStorage.kt

```diff
@@ -59,15 +59,23 @@ class TabCollectionStorage(
         }
     }
 
-    suspend fun createCollection(title: String, sessions: List<TabSessionState>) = ioScope.launch {
-        val id = collectionStorage.createCollection(title, sessions)
-        notifyObservers { onCollectionCreated(title, sessions, id) }
-    }.join()
+    suspend fun createCollection(title: String, sessions: List<TabSessionState>): Long? {
+        var id: Long? = null
```

Comment:
I feel like this may bite us in the back later on down the line, but right now, I don't know why the compiler is failing us.

Reply:
I'll try to see if there's a better way to do this.